quinn-proto/src/crypto.rs

```diff
@@ -186,3 +188,16 @@ pub trait HmacKey: Send + Sized + Sync {
     /// Method for verifying a message
     fn verify(&self, data: &[u8], signature: &[u8]) -> Result<(), ()>;
 }
+
+/// A key for sealing data with AEAD-based algorithms
+pub trait AeadKey {
+    /// Length of AEAD Key
+    const KEY_LEN: usize;
+
+    /// Method for deriving a fixed AEAD key using a `master_key` and `random_bytes`
+    fn derive_fixed(master_key: &[u8], random_bytes: &[u8]) -> Self;
+    /// Method for sealing message `data`
+    fn seal(&mut self, data: &mut Vec<u8>) -> Result<(), ()>;
+    /// Method for opening a sealed message `data`
+    fn open<'a>(&mut self, data: &'a mut Vec<u8>) -> Result<&'a mut [u8], ()>;
```

Comment:
```suggestion
    fn open<'a>(&mut self, data: &'a mut [u8]) -> Result<&'a mut [u8], ()>;
```
This doesn't need to be a `Vec`, and using a slice lets us save an alloc and copy.